### PR TITLE
Advanced Fatique - Take height difference into account

### PR DESCRIPTION
**Pull request by ulteq**
* Loosely based on: https://github.com/acemod/ACE3/pull/4492
* Closes #4492

-> A little more fatigue when walking uphill.
-> Less fatigue when walking downhill.
-> Similar fatigue when moving along the contour line

### CODE REVIEW

**Comment by bux** [comment]
> Less fatigue when walking downhill.

can't wait for that

**Comment by commy2** [comment]
Where does `GVAR(terrainGradientFactor)` come from?

**Comment by ulteq** [comment]
It's part of the original code. I think it comes from ACE_Settings.

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf [hunk 1]
```diff
@@ -16,12 +16,14 @@
...
  */
 #include "script_component.hpp"
-params ["_unit", "_velocity"];
+params ["_unit", "_movementSpeed"];
 
 private _gearMass = ((_unit getVariable [QEGVAR(movement,totalLoad), loadAbs _unit]) / 22.046) * GVAR(loadFactor);
 
-private _terrainAngle = asin (1 - ((surfaceNormal getPosASL _unit) select 2));
-private _terrainGradient = (_terrainAngle / 45 min 1) * 5 * GVAR(terrainGradientFactor);
+private _surfaceNormal = surfaceNormal (getPosASL _unit);
```
Comment (commy2) [inline addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: `getPosWorld`. Same X,Y; but skips the Z offset from model center to LandContact.

**Comment by ulteq** [comment]
My terrain gradient calculation is unfinished though.

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf [hunk 2]
```diff
@@ -32,20 +34,29 @@ private _duty = GVAR(animDuty);
     };
 } forEach (GVAR(dutyList) select 1);
 
-if (GVAR(isSwimming)) then {
-    _terrainGradient = 0;
+if (!GVAR(isSwimming)) then {
+    private _movementVector = vectorNormalized (velocity _unit);
+    private _side = vectorNormalized(_surfaceNormal vectorCrossProduct [0, 0, 1]);
+    private _up = vectorNormalized(_surfaceNormal vectorCrossProduct _side);
```
Comment (commy2) [inline addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: Would love if we had debug lines just like the grenade throwing for this.

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf [hunk 3]
```diff
@@ -32,20 +34,29 @@ private _duty = GVAR(animDuty);
...
+    private _upGradient = -(_movementVector vectorDotProduct _up) * _terrainGradient;
+    private _sideGradient = abs(_movementVector vectorDotProduct _side) * _terrainGradient;
```
Comment (commy2) [inline addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: I don't think this is right.

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf (hunk 2 above)
Comment (ulteq) [reply addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: I have them locally. Let me add them.

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf (hunk 3 above)
Comment (ulteq) [reply addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: It's not. I fixed it.

**Comment by ulteq** [comment]
Should be ready for review now.

**Comment by ulteq** [comment]
The Metabolic Cost Equation is still incorrect.

**Comment by ulteq** [comment]
Closed in favor of: https://github.com/acemod/ACE3/pull/5723

**Pull request closed without merging**